broker/index.js

```diff
@@ -34,323 +36,157 @@ var debug = function() {
 }
 
 // ############### Sensors communication ######################
-
-var tcpServerForSensors = net.createServer(function(tcpSocketSensor) {
-
-    console.log("New socket to sensor.");
-    var sim;
-
-    var tcpSocketSensorReceiver = makeTcpReceiver(tcpSocketSensor, "\n");
-    
-    tcpSocketSensorReceiver.on('message', function(message) {
-
-        console.log("received tcp data: ", message);
-
-        // register the sim id corresponding to the socket
-        var match = message.match(/init\s(\d+)\s(.+)/)
-        if (match) {
-            var token = match[2];
-            if (token === PRIVATE.token){
-                sim = match[1];
-                sim2socket[sim] = tcpSocketSensor;
-                console.log(tcpSocketSensor.remoteAddress + " is now known as " + sim);
-                
-                // check sensor is registered
-                database.Sensors.findBySIMid(sim)
-                .then(function(sensors) {
-                    // if sensor was never registered, create it
-                    if(sensors.length === 0) {
-                        console.log("Creating the sensor")
-                        return database.Sensors.create({
-                            'name': pokemon.random(),
-                            'sim': sim,
-                            'data_period': CONF.data_period,
-                            'start_time': CONF.start_time,
-                            'stop_time': CONF.stop_time
-                        })
-                        .then(function(createdSensor){
-                            return createdSensor;
-                        })
-                        .catch(function(err) {
-                            console.log("[ERROR] Couldn't create sensor :", err);
-                        })
-                    }
-                    else { 
-                        return sensors[0];
-                    }
-                    
+var authenticate = function(client, username, token, callback) {
+    var authorized = (token.toString() === PRIVATE.token);
+    if (authorized) {
+
+        // check if sensor exists in db
+        database.Sensors.get(client.id)
```

Comment:
Shouldn't we call `client.id` -> `client.sim`for the sake of clarity ?? I guess that the `client` should send its `id` under `sim`.